config.go

```diff
@@ -286,6 +286,11 @@ func NewSettings(filename string) (*Settings, error) {
 	if err != nil {
 		return nil, err
 	}
+
+	if err := settings.Validate(); err != nil {
```

Comment:
yep, a lot of integration tests are not getting past the validation

[2017/08/17 16:21:44.975] --- FAIL: TestCLIFetchSource (0.00s)
[2017/08/17 16:21:44.977] panic: Error opening settings override file /data/mci/83081e45cbd076b6dba17762435b192e/gopath/src/github.com/evergreen-ci/evergreen/creds.yml: DBUrl and DB must not be empty [recovered]
[2017/08/17 16:21:44.977] 	panic: Error opening settings override file /data/mci/83081e45cbd076b6dba17762435b192e/gopath/src/github.com/evergreen-ci/evergreen/creds.yml: DBUrl and DB must not be empty